ca_tools/scripts/draw_square.py

```diff
@@ -0,0 +1,214 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+import curses
+import math
+
+import rospy
+from geometry_msgs.msg import Twist, Pose, Quaternion, PoseWithCovariance
+from nav_msgs.msg import Odometry
+from tf.transformations import euler_from_quaternion as efq
+from tf.transformations import quaternion_from_euler as qfe
+
+ANGLE_THRESHOLD = 0.05
+ANGULAR_VEL = 0.3
+LINEAR_VEL = 0.5
+LENGTH_THRESHOLD = 0.05
+GOAL_DIST_THRESHOLD = 0.05
+
+
+class RobotState():
+
+    STATES = [
+    "MOVE_FORWARD",
+    "ROTATE",
+    "STOP"
+    ]
+
+    _current_state = None
+
+    def __init__(self):
+        self._current_state = "STOP"
+
+    def SetState(self,state):
+        "assert a valid state"
+        self._current_state = state
+
+    def GetState(self):
+        return self._current_state
+
+
+class GTSys():
+
+    _pose = Pose()
+
+    def __init__(self):
+        self._sub_gts=rospy.Subscriber('gts', Odometry, self._callback)
+
+    def _callback(self,data):
+        self._pose = data.pose
+
+    def GetPose(self):
+        return self._pose
+
+
+
+class DrawSquare():
+
+
+    _current_pose = PoseWithCovariance()
+    _state = RobotState()
+    _twist = Twist()
+    _next_pose = Pose()
+    _ground_truth = GTSys()
+    _goal_angle = 0
+    _currentGoalReached = False
+    _goal_pose = Pose()
+    _diff_angle = 0
+    _length_diff = 0
+    _square_length = 2
+
+
+    def __init__(self):
+        self._pub_cmd = rospy.Publisher('cmd_vel', Twist, queue_size=10)
+        self._sub_cmd = rospy.Subscriber('cmd_vel', Twist, self._callback)
+        self._hz = rospy.get_param('~hz', 10)
+        self._current_pose = self._ground_truth.GetPose()
+        self._rate = rospy.Rate(self._hz)
+        self._rate.sleep()
+
+
+    def _wrapAngle(self,angle): #ToDo: make beautiful this horrible function
+        while (angle<0):
+            angle+=2.0*math.pi
+        while (angle>2*math.pi):
+            angle-=2.0*math.pi
+        return angle
+
+
+    def _publish(self):
```

Comment:
Is this method necessary? Since it's doing only one thing, consider avoiding it.